ts/twitter/src/app/posts/[id]/_components/use-post-detail.ts

```diff
@@ -0,0 +1,38 @@
+"use client";
+
+import { useRef, useState } from "react";
+
+enum FollowState {
+  Follow,
+  UnFollow,
+}
+
+interface usePostDetailReturn {
+  followDisplayText: string;
+  toggleFollowState: () => void;
+}
+
+export const usePostDetail = (): usePostDetailReturn => {
+  const follow = useRef<FollowState>(FollowState.UnFollow);
+  const [followDisplayText, setFollowDisplayText] = useState<string>("follow");
```

Comment:
"setFollowDisplayText" -> "setFollowButtonText"

```suggestion

export const usePostDetail = (): usePostDetailReturn => {
  const follow = useRef<FollowState>(FollowState.UnFollow);
  const [followDisplayText, setFollowButtonText] = useState<string>("follow");
```